Practical 1 Scientific Calculator.

Jay-Prajapat opened the pull request:
completed practical 1 of scientific calculator.

Jay-Prajapat committed:
Java Script Practical-1 Started

Completed UI part and some backend functionality.

Jay-Prajapat committed:
implemented trigonometry functions, memory part and other functions.

Jay-Prajapat committed:
make changes in css of second button on onclick  event.

Jay-Prajapat committed:
remove commented and unnecessary code.

Jay-Prajapat committed:
updated script.js file

Jay-Prajapat committed:
path changed in calc.html in head section.

Jay-Prajapat committed:
implemented pi function that is forget by mistake and remove  inline css from index.html

dilaara-simform commented on Practicals/Practical1/calc.html:
```diff
@@ -0,0 +1,254 @@
+      crossorigin="anonymous"
+    ></script>
+    <script
+      src="https://cdn.jsdelivr.net/npm/@popperjs/core@2.9.2/dist/umd/popper.min.js"
```
what is the use of this JS ?

dilaara-simform commented on Practicals/Practical1/calc.html:
```diff
@@ -0,0 +1,254 @@
+                </ul>
+                <ul>
+                  <button onclick="random()">rand</button>
+                  <button data-operation>→dms</button>
```
what does data-operation do ?

dilaara-simform commented on Practicals/Practical1/js/script.js:
```diff
@@ -0,0 +1,329 @@
+for (item of opr) {
+  item.addEventListener("click", (e) => {
+    oprText = e.target.innerText;
+    console.log(oprText);
```
console.log should be removed before pushing the code.

dilaara-simform commented on Practicals/Practical1/js/script.js:
```diff
@@ -0,0 +1,329 @@
+  item.addEventListener("click", (e) => {
+    oprText = e.target.innerText;
+    console.log(oprText);
+    if (oprText == "x") {
```
implement this if cases using switch case or else if ladder

Jay-Prajapat committed:
Remove console logs for code optimization from script.js file

Jay-Prajapat committed:
change in  script.js file and remove javascript link from calc.html file.

change multiple if statement to else if ladder in script.js file

dilaara-simform commented on Practicals/Practical1/js/script.js:
```diff
@@ -0,0 +1,322 @@
+var screen = document.querySelector(".current");
+var ans = document.querySelector(".previous");
+var btn = document.querySelectorAll(".numbtn");
+var opr = document.querySelectorAll(".operator");
```
you should not use class to access the values.

dilaara-simform commented on Practicals/Practical1/js/script.js:
```diff
@@ -0,0 +1,322 @@
+    let lastCharacter = ans.innerHTML[ans.innerHTML.length - 1];
+
+    if (operators.includes(lastCharacter)) {
+      ans.innerHTML = ans.innerHTML.slice(0, -1) + oprText;
```
create a common function to update innerHTML

dilaara-simform commented on Practicals/Practical1/js/script.js:
```diff
@@ -0,0 +1,322 @@
+let memory = [];
+
```
Numeric input from the keyboard should be allowed.

dilaara-simform commented on Practicals/Practical1/calc.html:
```diff
@@ -0,0 +1,249 @@
+      </div>
+      <div class="row1">
+        <div class="col" id="xPowy">
+          <button onclick="pow()">x<sup>y</sup></button>
```
multiple clicks should not be allowed.

Jay-Prajapat committed:
Implemented changes that are given in comment.

Use id instedof class in queryselector, created common function to update innerhtml, allowed numeric input from the keyboard, implemented multiple click that not allowd.
